lib/create-contentful-api.ts

```diff
@@ -30,19 +30,19 @@ export interface ContentfulClientApi {
 
   getAsset(id: string): Promise<Asset>
 
-  getAssets(query?: FieldsQueries<AssetFields>): Promise<AssetCollection>
+  getAssets(query?: AssetQueries<AssetFields>): Promise<AssetCollection>
 
   getContentType(id: string): Promise<ContentType>
 
   getContentTypes(): Promise<ContentTypeCollection>
 
   getEntries<Fields = Record<string, any>>(
-    query?: FieldsQueries<Fields>
+    query?: EntryQueries<Fields>
   ): Promise<EntryCollection<Fields>>
 
   getEntry<Fields = Record<string, any>>(
     id: string,
-    query?: FieldsQueries<Fields>
+    query?: EntryQueries<Fields>
```

Comment:
I am still confused what kind of query we would send together with fetching an entry by ID.
Things like `select`?